bundle/edu.gemini.obslog/src/main/scala/edu/gemini/obslog/instruments/GNIRSLogSegment.scala

```diff
@@ -0,0 +1,154 @@
+package edu.gemini.obslog.instruments
+
+import java.util.logging.Logger
+
+import edu.gemini.obslog.config.model.OlLogItem
+import edu.gemini.obslog.core.OlSegmentType
+import edu.gemini.obslog.obslog.{ConfigMap, InstrumentLogSegment, OlLogOptions}
+import edu.gemini.spModel.gemini.gnirs.{GNIRSParams, InstGNIRS}
+import edu.gemini.spModel.obscomp.InstConstants
+
+import GNIRSParams._
+import GNIRSLogSegment._
+
+import scalaz._
+import Scalaz._
+
+class GNIRSLogSegment(logItems: java.util.List[OlLogItem], obsLogOptions: OlLogOptions) extends InstrumentLogSegment(SegmentType, logItems, obsLogOptions) {
+
+  override def decorateObservationData(map: ConfigMap): Unit = Option(map).foreach(m => {
+    decorateFilter(m)
+    decorateSlitWidth(m)
+    decorateDisperser(m)
```

Comment:
Ooops. I wrote the code for it, but neglected to add it to the list. Good catch.
As I said, this is untested: I finished it late Tuesday night and just wanted to throw it up - working or not - to see what people thought.